packages/frontend/src/pages/NewListing.tsx

```diff
@@ -0,0 +1,445 @@
+              <FormControl>
+                <Input
+                  {...field}
+                  type='text'
+                  value={displayPrice}
+                  className='w-[180px] text-base'
+                  onChange={(e) => {
+                    const value = parseFloat(e.target.value.replace(/[^\d\.]/g, ''));
+                    if (!isNaN(value)) {
+                      setValue('price', value);
+                      setDisplayPrice(e.target.value)
+                    }
+                  }}
+                  onBlur={async () => {
+                    await trigger('price');
+                    field.onBlur();
+                    setDisplayPrice(new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD' }).format(price));
+                  }}
+                />
+              </FormControl>
+              {errors && <FormMessage>{errors.price?.message}</FormMessage>}
+            </FormItem>
+          )
+        }}
+      />
+      <FormField
+        control={control}
+        name="description"
+        render={({ field }) => (
+          <FormItem>
+            <FormLabel className='text-base'>Description</FormLabel>
+            <FormControl>
+              <Textarea className='text-base'
+                {...field}
+                onBlur={async (e) => {
+                  await trigger('description');
+                  field.onBlur();
+                }}
+              />
+            </FormControl>
+            {errors && <FormMessage>{errors.description?.message}</FormMessage>}
+          </FormItem>
+        )}
+      />
+    </section >
+  )
+}
+
+type TrustScoreFormStepProps = StepSectionProps & { trustScores: Record<TrustScoreKey, TrustScoreInfo> }
+
+const TrustScoreFormStep = ({ control, trustScores: trustScoresFromData }: TrustScoreFormStepProps) => {
+  return (
+    <section className='flex flex-col space-y-4'>
+      {Object.entries(trustScoresFromData).map(([key, scoreInfo]) => (
+        <FormField
+          key={key}
+          control={control}
+          name={`revealTrustScores.${key}`}
+          render={({ field }) => (
+            <FormItem className='flex justify-between items-start space-x-6 border border-muted-foreground p-3' key={key}>
+              <div>
+                <FormLabel className="text-foreground text-lg" htmlFor={key}>
+                  {scoreInfo.title}:{' '}<span className="text-blue-700 font-extrabold">{scoreInfo.score}%</span>
+                </FormLabel>
+                <FormDescription className='text-foreground/80'>{scoreInfo.description}</FormDescription>
+              </div>
+              <FormControl>
+                <div className="flex space-x-2">
+                  <Switch
+                    id={key}
+                    checked={field.value}
+                    onCheckedChange={field.onChange}
+                    className={cn(field.value ? 'data-[state=checked]:bg-blue-700' : '')}
+                  />
```

Comment:
@thebeyondr the Switch doesn't appear to change `isRevealed` to false. when i choose hide a scores from the ui,  are still sent on the listing. i don't understand where `field.value` and `field.onChange` come from, where are we asking `isRevealed` to flip from T/F ?
the logic to create score obj to send to the db (only add score if `isRevealed`) is next comment 👇

Reply:
Yeah it doesn't use `isRevealed`. its connected to `revealTrustScores` (see line 246)